clients/drcachesim/tools/func_view.cpp

```diff
@@ -335,5 +358,20 @@ func_view_t::print_results()
     return true;
 }
 
+func_view_t::traced_info_t &
+func_view_t::get_info_for_last_func_id(shard_data_t *shard)
+{
+    assert(shard != nullptr);
+    int id = shard->last_func_id;
+    assert(id != -1);
+    traced_info_t &info = id2info_[id];
+    // If this entry is uninitialized, it means we aren't expecting to see this
```

Comment:
Isn't the normal pattern to use id2info_.find() to see whether the entry is there, rather than always creating something and then seeing if we just created it?

Reply:
Originally I did this to detect previously-created but uninitialized entries, since the original code quietly created missing entries, but now that all other uses of id2info_[] are either gated or in the final print, it's probably cleaner to do it the normal way.  Updated.